core/include/kernel/notif.h

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Linaro Limited
+ */
+
+#ifndef __KERNEL_NOTIF_H
+#define __KERNEL_NOTIF_H
+
+#include <compiler.h>
+#include <sys/queue.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+
+/*
+ * Notification values are divided into two kinds, asynchronous and
+ * synchronous, where the asynchronous has the lowest values.
+ * They are ordered as:
+ * 0			    Do bottom half
+ * 1..NOTIF_ASYNC_MAX	    Free for signalling in PTAs and should be
+ *			    allocated with notif_alloc_async_value()
+ * NOTIF_SYNC_VALUE_BASE..  Used as NOTIF_SYNC_VALUE_BASE + thread_id
+ * NOTIF_VALUE_MAX	    for mutex and condvar wait/wakeup
+ *
+ * Any value can be signalled with notif_send_sync() while only the ones
+ * <= CFG_CORE_ASYNC_NOTIF_MAX can be signalled with notif_send_async().
```

Comment:
When calling `notif_send_sync()` with a value in the async range, is it still handled synchronously? (I suppose it is, I just want to make sure).

Reply:
Yes, I'll update the description of notif_send_sync().